libs/core/src/lib/combobox/combobox.component.ts

```diff
@@ -705,10 +701,12 @@ export class ComboboxComponent
 
     /** @hidden */
     private _propagateChange(): void {
-        if (!this.communicateByObject) {
-            this.onChange(this.inputText);
-        } else {
+        if (this.communicateByObject) {
+            const value = this._getOptionObjectByDisplayedValue(this.inputText);
```

Comment:
If we want to communicate by object, we don't want to just get the object by the displayed value. Take a look at the reactive example, which uses these two values:
```
        { displayedValue: 'Apple', value: 'AppleValue' },
        { displayedValue: 'Apple', value: 'AppleValue2' },
```
Now we can no longer select the option with AppleValue2. You'll see this is working on main